lib/Net/Async/Blockchain/Omni.pm

```diff
@@ -127,21 +155,32 @@ async sub transform_transaction {
 
     my $transaction = Net::Async::Blockchain::Transaction->new(
         currency     => $self->currency_symbol,
-        hash         => $decoded_raw_transaction->{txid},
+        hash         => $omni_transaction->{txid},
         block        => $block,
         from         => $from->{address},
         to           => $to->{address},
         amount       => $amount,
         fee          => $fee,
         fee_currency => FEE_CURRENCY,
         type         => $transaction_type,
-        property_id  => $received_transaction->{propertyid},
-        timestamp    => $received_transaction->{blocktime},
+        property_id  => $omni_transaction->{propertyid} // PROPERTY_ID,
+        timestamp    => $omni_transaction->{blocktime},
     );
 
-    $self->source->emit($transaction) if $transaction;
+    return $transaction if $transaction;
 
-    return 1;
+}
+
+=head2 mapping_address
+
+This will return from and to hash.
+
+=cut
+
+sub mapping_address {
+
+    my ($self, $omni_transaction) = @_;
+    return Future->needs_all(map { $self->rpc_client->validate_address($omni_transaction->{$_}) } qw(sendingaddress referenceaddress))->get;
```

Comment:
why are you  blocking in async code `->get`?

Reply:
Instead of using ```await``` , was trying to use ```->get``` .. Just probably checking...  comments are addressed